xarray/core/dataarray.py

```diff
@@ -244,6 +240,16 @@ def _replace_maybe_drop_dims(self, variable, name=__default):
                                  if set(v.dims) <= allowed_dims)
         return self._replace(variable, coords, name)
 
+    def _replace_indexes(self, indexes):
+        obj = self
+        for dim, idx in iteritems(indexes):
+            if idx.name is None:
+                idx.name = dim + "_unnamed_level"
```

Comment:
IMO `level_0`, `level_1`, etc. is the best option. I finally found a way to handle this: set default levels names on a copy of the MultIndex before calling `convert_label_indexer`.